resources/qml/dialogs/RoomSettings.qml

```diff
@@ -152,171 +169,222 @@ ApplicationWindow {
-                roomid: roomSettings.roomId
-                roomName: roomSettings.roomName
-            }
+                Button {
+                    text: qsTr("Configure")
+                    ToolTip.text: qsTr("Select events to hide in this room")
+                    onClicked: hiddenEventsDialog.show()
+                    Layout.alignment: Qt.AlignRight
+                }
 
-            Button {
-                text: qsTr("Configure")
-                ToolTip.text: qsTr("Select events to hide in this room")
-                onClicked: hiddenEventsDialog.show()
-                Layout.alignment: Qt.AlignRight
-            }
+                Item {
+                    // for adding extra space between sections
+                    Layout.fillWidth: true
+                }
 
-            Item {
-                // for adding extra space between sections
-                Layout.fillWidth: true
-            }
+                Item {
+                    // for adding extra space between sections
+                    Layout.fillWidth: true
+                }
 
-            Item {
-                // for adding extra space between sections
-                Layout.fillWidth: true
-            }
+                Label {
+                    text: qsTr("INFO")
+                    font.bold: true
+                    color: Nheko.colors.text
+                }
 
-            MatrixText {
-                text: qsTr("INFO")
-                font.bold: true
-            }
+                Item {
+                    Layout.fillWidth: true
+                }
 
-            Item {
-                Layout.fillWidth: true
-            }
+                Label {
+                    text: qsTr("Internal ID")
+                    color: Nheko.colors.text
+                }
 
-            MatrixText {
-                text: qsTr("Internal ID")
-            }
+                AbstractButton { // AbstractButton does not allow setting text color
+                    Layout.alignment: Qt.AlignRight
+                    Layout.fillWidth: true
+                    Layout.preferredHeight: idLabel.height
+                    Label { // TextEdit does not trigger onClicked
+                        id: idLabel
+                        text: roomSettings.roomId
+                        font.pixelSize: Math.floor(fontMetrics.font.pixelSize * 0.8)
+                        color: Nheko.colors.text
+                        width: parent.width
+                        horizontalAlignment: Text.AlignRight
+                        wrapMode: Text.WrapAnywhere
+                        ToolTip.text: qsTr("Copied to clipboard")
+                        ToolTip.visible: toolTipTimer.running
+                    }
+                    TextEdit{ // label does not allow selection
+                        id: textEdit
+                        visible: false
+                        text: roomSettings.roomId
+                    }
+                    onClicked: {
```

Comment:
This feels a bit weird on the desktop. It feels like we are mixing two paradigms. But I guess you rarely want to copy part of the room id...